### PR TITLE
Fix class loading issue with single bal file with interop

### PR DESCRIPTION
**Pull request by warunalakshitha**
## Purpose
This will add module based class loading when building the ballerina program. The required changes has been done for gradle build as well.

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/19497

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/CopyNativeLibTask.java [hunk 1]
```diff
@@ -202,31 +185,49 @@ private static Path findImportBaloPath(BuildContext buildContext, BPackageSymbol
...
-                    }
-                    // get the input stream
-                    try (InputStream is = jar.getInputStream(file)) {
-                        Files.copy(is, f.toPath());
+        String fileName = baloFilePath.getFileName().toString();
+        Path baloFileUnzipDirectory = Paths.get(baloFilePath.getParent().toString(),
+                                                fileName.substring(0, fileName.lastIndexOf(".")));
+        File destFile = baloFileUnzipDirectory.toFile();
+        // Read from .balo file if directory not exist.
+        if (destFile.mkdir()) {
```
Comment (Kishanthan) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/CopyNativeLibTask.java]: Shall we revers the if here?

File: compiler/ballerina-backend-jvm-old/src/main/ballerina/src/compiler_backend_jvm/main.bal [hunk 2]
```diff
@@ -31,21 +32,31 @@ public type JavaClass record {|
...
-    var numCacheDirs = args.length() - 4;
     int i = 0;
     while (i < numCacheDirs) {
-        birCacheDirs[i] = args[4 + i];
+        birCacheDirs[i] = <@untainted> args[6 + i];
         i = i + 1;
     }
 
-    var jarFile = generateJarBinary(pathToEntryBir, mapPath, dumpBir);
+    int argsCount = 6 + numCacheDirs;
```
Comment (Kishanthan) [inline compiler/ballerina-backend-jvm-old/src/main/ballerina/src/compiler_backend_jvm/main.bal]: shall we put a comment about number 6 here?

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/CreateJarTask.java [hunk 3]
```diff
@@ -110,10 +123,25 @@ private void writeImportJar(Path tmpDir, List<BPackageSymbol> imports, Path sour
...
+        }
+    }
+
+    private Path getRuntimeAllJar(BuildContext buildContext) {
+
+        if (skipCopyLibsFromDist) {
+            return null;
         }
+        String balHomePath = buildContext.get(BuildContextField.HOME_REPO).toString();
+        String ballerinaVersion = System.getProperty("ballerina.version");
```
Comment (hemikak) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/CreateJarTask.java]: Use RepoUtils.getBallerinaVersion() here

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/buildcontext/BuildContext.java [hunk 4]
```diff
@@ -68,6 +70,8 @@
     private SourceType srcType;
     private transient PrintStream out;
     private transient PrintStream err;
+
```
Comment (hemikak) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/buildcontext/BuildContext.java]: Shall we remove the empty line.

**Comment by Kishanthan** [comment]
This will also address https://github.com/ballerina-platform/ballerina-lang/issues/19160

**Pull request merged**